### PR TITLE
fix: handle wait non-wait properly in testing

### PR DESCRIPTION
**Pull request by LarsLiden**
also lowered re-prompt threshold

### CODE REVIEW

File: src/http/router.ts [hunk 1]
```diff
@@ -1218,46 +1218,48 @@ export const getRouter = (client: CLClient, options: ICLClientOptions): express.
                         validity = CLM.TranscriptValidationResultType.CHANGED
                     }
 
-                    // Now take the actual action to update memory 
+                    // Get conversation reference
                     const conversationReference = BB.TurnContext.getConversationReference(activity)
                     if (!conversationReference) {
                         validity = CLM.TranscriptValidationResultType.TEST_FAILED
+                        break
```
Comment (tpmsr) [inline src/http/router.ts]: :)

File: src/http/router.ts [hunk 2]
```diff
@@ -1218,46 +1218,48 @@ export const getRouter = (client: CLClient, options: ICLClientOptions): express.
...
-                    else {
+
+                    // Get session Id
+                    const sessionId = await result.memory.BotState.GetSessionIdAsync()
+                    if (!sessionId) {
+                        validity = CLM.TranscriptValidationResultType.TEST_FAILED
+                        break
+                    }
+
+                    if (conversationReference && sessionId) {
```
Comment (tpmsr) [inline src/http/router.ts]: This check is now spurious with the above, right ?

File: src/http/router.ts [hunk 3]
```diff
@@ -1218,46 +1218,48 @@ export const getRouter = (client: CLClient, options: ICLClientOptions): express.
...
-                                    if (turnValidation.actionHashes[i] && !Utils.actionHasHash(bestAction.actionId, turnValidation.actionHashes[i], appDefinition.actions)) {
-                                        validity = CLM.TranscriptValidationResultType.CHANGED
-                                    }
-                                    // Include apiResults when taking action so result will be the same when testing
-                                    await clRunner.TakeActionAsync(conversationReference, result, null, turnValidation.apiResults[i])
-                                }
+                        // Trigger next action if non-terminal
+                        let bestAction = result.scoredAction
+                        let curHashIndex = 0
+                        while (!bestAction.isTerminal) {
```
Comment (tpmsr) [inline src/http/router.ts]: Is it guaranteed that there will be a terminal action ?

Comment (LarsLiden) [reply src/http/router.ts]: Not sure what you mean.

Comment (tpmsr) [reply src/http/router.ts]: Rephrasing - is there a possibility that this will loop infinitely ?

Comment (LarsLiden) [reply src/http/router.ts]: Ah, gotcha.  The server has a max number of non-terminal actions that it allows.  I'll add a comment

**Pull request merged**